zookeeper/source/zk_nn_client.cc

```diff
@@ -1476,7 +1475,7 @@ u_int64_t ZkNnClient::generate_block_group_id() {
 }
 
 u_int64_t ZkNnClient::get_block_group_id(u_int64_t storage_block_id) {
-    u_int64_t mask = ((1ull << 47) - 1) << 16;  // 47 ones and 16 zeros.
+    u_int64_t mask = (~(0ull) << 16);
```

Comment:
why was this change needed?

Reply:
It was giving the wrong value, I think because  1ull << 47 - 1 doesn't give all 1's.